File: web-runtime/volunteer-pages/yuzan-volunteer-training-ui/yuzan-volunteer-training-ui/scripts/screenshot.mjs
import { chromium } from 'playwright';
import { readFile } from 'node:fs/promises';
import { extname } from 'node:path';
const browser=await chromium.launch({headless:true,executablePath:'/usr/bin/chromium',args:['--no-sandbox','--disable-gpu']});
const page=await browser.newPage({viewport:{width:1672,height:941},deviceScaleFactor:1});
let html=await readFile('index.html','utf8');
let css=await readFile('styles.css','utf8');
let js=await readFile('app.js','utf8');
const assets=['logo-symbol.png','sidebar-mountain.jpg','header-ridge.jpg','path-landscape.jpg','book-valley.png','learning-tree.png','teacher-book.png','network-valley.png'];
for(const name of assets){
  const buf=await readFile(`assets/${name}`); const ext=extname(name).slice(1).replace('jpg','jpeg'); const uri=`data:image/${ext};base64,${buf.toString('base64')}`;
  html=html.split(`assets/${name}`).join(uri); css=css.split(`assets/${name}`).join(uri); js=js.split(`assets/${name}`).join(uri);
}
html=html.replace('<link rel="stylesheet" href="styles.css" />',`<style>${css}</style>`).replace('<script src="app.js"></script>',`<script>${js}</script>`);
await page.setContent(html,{waitUntil:'load'});
await page.waitForTimeout(700);
await page.screenshot({path:'qa/current.png',fullPage:true});
await browser.close();

File: web-runtime/student-pages/yuzan-student-exercise-ui/yuzan-student-exercise-ui/scripts/screenshot.py
from pathlib import Path
from base64 import b64encode
from playwright.sync_api import sync_playwright
import re

root=Path(__file__).resolve().parents[1]
html=(root/'index.html').read_text(encoding='utf-8')
css=(root/'styles.css').read_text(encoding='utf-8')
js=(root/'app.js').read_text(encoding='utf-8')

def data_uri(path: Path):
    ext=path.suffix.lower()
    mime={'.png':'image/png','.jpg':'image/jpeg','.jpeg':'image/jpeg','.svg':'image/svg+xml'}[ext]
    return f'data:{mime};base64,'+b64encode(path.read_bytes()).decode()

for name in ['logo-symbol.png','header-ridge.png','student-avatar.jpg','teacher-avatar.jpg','paper-noise.png','grass.svg']:
    uri=data_uri(root/'assets'/name)
    css=css.replace(f'assets/{name}',uri)
    html=html.replace(f'assets/{name}',uri)
html=re.sub(r'<link rel="stylesheet" href="styles\.css"\s*/?>',f'<style>{css}</style>',html)
html=html.replace('<script src="app.js"></script>',f'<script>{js}</script>')

shots=[('desktop.png',1659,948),('tablet.png',768,1200),('mobile.png',390,1300)]
with sync_playwright() as pw:
    browser=pw.chromium.launch(headless=True,executable_path='/usr/bin/chromium',args=['--no-sandbox','--disable-dev-shm-usage'])
    for name,w,h in shots:
        page=browser.new_page(viewport={'width':w,'height':h},device_scale_factor=1)
        page.set_content(html,wait_until='load')
        page.wait_for_timeout(350)
        page.screenshot(path=str(root/'qa'/name),full_page=(name != 'desktop.png'))
        page.close()
    browser.close()
print('screenshots created')
